pkgs/tools/typesetting/marp/default.nix

```
@@ -0,0 +1,34 @@
+{
+  mkYarnPackage
+, fetchFromGitHub
+, lib
+}:
+
+let version = "3.1.0";
+in
+mkYarnPackage {
+  name = "marp-cli";
+  version = version;
+  src = fetchFromGitHub {
+    owner = "marp-team";
+    repo = "marp-cli";
+    rev = "v${version}";
+    sha256 = "2TsiHv598uxbns8CAz5mljSsFU3gFuHaZpBaaE2KMkE=";
+  };
+
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
```

You can actually get by without `yarn.lock` or `yarn.nix` (which are large-ish files) via something like this:

```suggestion
  offlineCache = fetchYarnDeps {
    yarnLock = "${src}/yarn.lock";
    hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  };
```

---

Replaced with 

```diff
--- a/pkgs/tools/typesetting/marp/default.nix
+++ b/pkgs/tools/typesetting/marp/default.nix
@@ -1,14 +1,13 @@
 {
   mkYarnPackage
+, fetchYarnDeps
 , fetchFromGitHub
 , lib
 }:

-let version = "3.1.0";
-in
-mkYarnPackage {
+mkYarnPackage rec {
   name = "marp-cli";
-  version = version;
+  version = "3.1.0";
   src = fetchFromGitHub {
     owner = "marp-team";
     repo = "marp-cli";
@@ -17,8 +16,10 @@ mkYarnPackage {
   };

   packageJSON = ./package.json;
-  yarnLock = ./yarn.lock;
-  yarnNix = ./yarn.nix;
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-TJ8fpnsbcSG2gYvQyCMWKMyMxaybVi3LB5Ea2F1elDw=";
+  };
```

got 

```
error: builder for '/nix/store/k0rmfjfr8km9drpikvmz3dj8dh5vxh3w-marp-team-marp-cli-modules-3.1.0.drv' failed with exit code 1;
       last 10 log lines:
       >     at Generator.next (<anonymous>)
       >     at step (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:310:30)
       >     at /nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:321:13
       > Error: Couldn't find any versions for "rollup" that matches "^3.26.3" in our cache (possible versions are ""). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.
       >     at MessageError.ExtendableBuiltin (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:721:66)
       >     at new MessageError (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:750:123)
       >     at NpmResolver.<anonymous> (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:50438:15)
       >     at Generator.next (<anonymous>)
       >     at step (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:310:30)
       >     at /nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:321:13
       For full logs, run 'nix log /nix/store/k0rmfjfr8km9drpikvmz3dj8dh5vxh3w-marp-team-marp-cli-modules-3.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/g6v5g8pclsvmm11ajpnjwhp0cc5vhs1h-marp-cli.drv' failed to build
```